languages/python/oso/oso/oso.py

```diff
@@ -42,24 +70,84 @@ def is_allowed(self, actor, action, resource) -> bool:
         except StopIteration:
             return False
 
-    def get_allowed_actions(self, actor, resource, allow_wildcard=False) -> list:
+    def get_allowed_actions(self, actor, resource, allow_wildcard=False) -> List[Any]:
+        """Determine the actions ``actor`` is allowed to take on ``resource``.
+
+        Deprecated. Use ``authorized_actions`` instead.
+        """
+        return list(self.authorized_actions(actor, resource, allow_wildcard))
+
+    def authorize(self, actor, action, resource, *, check_read=True):
+        """Ensure that ``actor`` is allowed to perform ``action`` on
+        ``resource``.
+
+        If the action is permitted with an ``allow`` rule in the policy, then
+        this method returns ``None``. If the action is not permitted by the
+        policy, this method will raise an error.
+
+        The error raised by this method depends on whether the actor can perform
+        the ``"read"`` action on the resource. If they cannot read the resource,
+        then a ``NotFound`` error is raised. Otherwise, a ``ForbiddenError`` is
+        raised.
+
+        :param actor: The actor performing the request.
+        :param action: The action the actor is attempting to perform.
+        :param resource: The resource being accessed.
+
+        :param check_read: If set to ``False``, a ``ForbiddenError`` is always
+            thrown on authorization failures, regardless of whether the actor can
+            read the resource. Default is ``True``.
+        :type check_read: bool
+
+        """
+        if self.query_rule_once("allow", actor, action, resource):
+            return
+
+        is_not_found = False
+        if check_read:
+            if action == self.read_action:
+                is_not_found = True
+            elif not self.query_rule_once("allow", actor, self.read_action, resource):
+                is_not_found = True
+        raise self.not_found_error() if is_not_found else self.forbidden_error()
```

Comment:
nit: I think this is cleaner than assigning to a boolean, but I don't mind if you prefer it as-is:

```suggestion
        if check_read and (
            action == self.read_action
            or not self.query_rule_once("allow", actor, self.read_action, resource)
        ):
            raise self.not_found_error()
        raise self.forbidden_error()
```